components/org.wso2.carbon.data.provider/src/main/java/org/wso2/carbon/data/provider/endpoint/DataProviderEndPoint.java

```diff
@@ -103,6 +124,42 @@ public static void onOpen(WebSocketConnection webSocketConnection) {
     @OnMessage
     public void onMessage(String message, WebSocketConnection webSocketConnection) {
         DataProviderConfigRoot dataProviderConfigRoot = new Gson().fromJson(message, DataProviderConfigRoot.class);
+        String authoringClassName;
+        try {
+            Map webSocketConfiguration
+                    = (Map)getDataProviderHelper().getConfigProvider().getConfigurationObject(WEB_SOCKET_CONFIG_HEADER);
+            if (webSocketConfiguration != null) {
+                JSONObject webSocketConfigJSON = new JSONObject(new Gson().toJson(webSocketConfiguration));
+                if (!webSocketConfigJSON.has(WEB_SOCKET_AUTHORIZING_CLASS_CONFIG_HEADER)) {
+                    throw new Exception("Web socket authorizing class cannot be found in the deployment.yaml file.");
+                }
+                authoringClassName = (String) webSocketConfigJSON.get(WEB_SOCKET_AUTHORIZING_CLASS_CONFIG_HEADER);
+            } else {
+                authoringClassName = DEFAULT_WEBSOCKET_AUTHORIZING_CLASS;
+            }
+            DataProviderAuthorizer dataProviderAuthorizer;
+            try {
+                dataProviderAuthorizer = getDataProviderHelper().getDataProviderAuthorizer(authoringClassName);
+            } catch (NullPointerException e) {
+                throw new Exception("Cannot find the Data Provider Authorizer class for the given class name: "
+                        + authoringClassName + ".");
+            }
```

Comment:
Can we avoid this. Rather than catching the NPE, can we check the condition and raise the exception at that point.

Reply:
Fixed in fd69aaa5391eed2876b72814600426b55d58ec1d